### PR TITLE
Set write watermark in keyserver

### PR DESCRIPTION
**Pull request by gdbelvin**
Add an interface to return metadata about writes to the mutation queue so it can be exposed in the keyserver as a metric.

### CODE REVIEW

File: impl/sql/mutationstorage/queue.go [hunk 1]
```diff
@@ -46,27 +47,32 @@ func (m *Mutations) AddLogs(ctx context.Context, directoryID string, logIDs ...i
...
 		if err != nil {
-			return err
+			return nil, err
 		}
 		updateData = append(updateData, data)
 	}
 	// TODO(gbelvin): Implement retry with backoff for retryable errors if
 	// we get timestamp contention.
-	return m.send(ctx, time.Now(), directoryID, logID, updateData...)
+	ts := time.Now()
```
Comment (pavelkalinnikov) [inline impl/sql/mutationstorage/queue.go]: Looks dangerous to me, as servers have time skew. Shouldn't the returned watermark be based on what the storage thinks the "time" is? Consider returning a "strong" watermark from the `send` call below.

Comment (gdbelvin) [reply impl/sql/mutationstorage/queue.go]: The write implementation first verifies that the timestamp provided is greater than any other timestamp currently in the database.  A server that has a slow clock will simply fail more often when racing another writer that has a clock further ahead.

Comment (gdbelvin) [reply impl/sql/mutationstorage/queue.go]: I provide the watermark here instead of returning it so that I can have meaningful unit tests.

Comment (pavelkalinnikov) [reply impl/sql/mutationstorage/queue.go]: Why does returning the same watermark that [this](https://github.com/google/keytransparency/blob/edd2c90f078bec0d40021e5c076e36c9899fb2e1/impl/sql/mutationstorage/queue.go#L138) line generates in `m.send` not work?

Comment (pavelkalinnikov) [reply impl/sql/mutationstorage/queue.go]: (or just the timestamp)

Comment (gdbelvin) [reply impl/sql/mutationstorage/queue.go]: That line doesn't query the current time, it just convertes the passed in `ts time.Time` into an integer `ts.UnixNano()`

Comment (pavelkalinnikov) [reply impl/sql/mutationstorage/queue.go]: Ah, indeed.

**Pull request merged**